src/main/java/de/tum/in/www1/artemis/service/ExamSubmissionService.java

```diff
@@ -125,4 +125,28 @@ private boolean isSubmissionInTime(Exercise exercise, StudentExam studentExam) {
         }
         return exam.getStartDate().isBefore(ZonedDateTime.now()) && calculatedEndDate.isAfter(ZonedDateTime.now());
     }
+
+    /**
+     * Checks if the current time is after the exercise due date (course exercises)
+     * Checks if the current time is after the latest exam end date (exam exercises)
+     * We need this check in order to see if the assessment of exercises can be started
+     * @param exercise
+     * @return boolean
+     */
+    public boolean checkIfExerciseDueDateIsReached(Exercise exercise) {
```

Comment:
I am wondering if this is the correct place for this method. I would prefer this to be part of a exam independent service.
The way this is currently used makes it hard to understand that this method does have relevance for non-exam exercises if not reading the comment (which I found hard to understand).

Reply:
@jpbernius I moved it to the SubmissionService :)